src/Vault.sol

```diff
@@ -234,7 +230,13 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver, ICDP, Multi
     /// @inheritdoc ICDP
     function getOverallDebt(uint256 vaultId) public view returns (uint256) {
         uint256 currentDebt = vaultDebt[vaultId];
-        return currentDebt + stabilisationFeeVaultSnapshot[vaultId] + _accruedStabilisationFee(vaultId, currentDebt);
+        uint256 deltaGlobalStabilisationFeeD = globalStabilisationFeePerUSDD() -
+            _globalStabilisationFeePerUSDVaultSnapshotD[vaultId];
+        return
+            currentDebt +
+            stabilisationFeeVaultSnapshot[vaultId] +
+            (currentDebt * deltaGlobalStabilisationFeeD) /
+            DENOMINATOR;
```

Comment:
Why `FullMath.mulDiv` isn't used?

Reply:
It seems that these multiplication cannot overflow, so there is no need to use FullMath..

However it probably makes more sense to review all type limits and overflow scenarios later, to see what can be optimized there, which types can be made shorter, etc.

Returned FullMath back in this PR for now